Delete NetEventSource.Fail

stephentoub opened the pull request:
At some point some Debug.Asserts/Fails were replaced by this NetEventSource.Fail helper, which both Debug.Fails and fires an EventSource event.  But asserts in our code base are intended for things that should never happen, and we needn't be emitting events for them (if we did want to emit events for them, we'd need to tackle the other ~20,000 Debug.Assert/Fails in the codebase.

I've deleted NetEventSource.Fail, and fixed up the call sites.  Some were simply replaced by Debug.Assert/Fail.  Some were deleted entirely, when from code inspection it looked like they could actually be hit, but were guarded by a check for the event source being enabled and thus were unlikely to have been triggered in our previous testing.  Etc.

MihaZupan commented on src/libraries/Common/src/System/Net/Security/SecurityBuffer.Windows.cs:
```diff
@@ -118,10 +111,7 @@ public SecurityBuffer(byte[]? data, SecurityBufferType tokentype)
-            {
-                NetEventSource.Fail(typeof(SecurityBuffer), $"'size' out of range.  [{size}]");
-            }
+            Debug.Assert(size < 0, $"'size' out of range.  [{size}]");
```
```suggestion
            Debug.Assert(size >= 0, $"'size' out of range.  [{size}]");
```

MihaZupan reviewed (approved):
Assert changes LGTM

stephentoub committed:
Delete NetEventSource.Fail

At some point some Debug.Asserts/Fails were replaced by this NetEventSource.Fail helper, which both Debug.Fails and fires an EventSource event.  But asserts in our code base are intended for things that should never happen, and we needn't be emitting events for them (if we did want to emit events for them, we'd need to tackle the other ~20,000 Debug.Assert/Fails in the codebase.

I've deleted NetEventSource.Fail, and fixed up the call sites.  Some were simply replaced by Debug.Assert/Fail.  Some were deleted entirely, when from code inspection it looked like they could actually be hit, but were guarded by a check for the event source being enabled and thus were unlikely to have been triggered in our previous testing.  Etc.
